torch/_inductor/compile_fx.py

```diff
@@ -665,78 +629,15 @@ def codegen_and_compile(
-                    BoxedBool.disable(cudagraphs)
-                else:
-                    complex_memory_overlap_inputs = any(
-                        complex_memory_overlap(t)
-                        for t in example_inputs
-                        if isinstance(t, torch.Tensor)
-                    )
-
-                    if not config.triton.cudagraph_support_input_mutation:
-                        # Skip supports for cudagraph-managed tensors
-                        from torch._inductor.cudagraph_utils import (
-                            check_for_mutation_ignore_cuda_graph_managed_tensor,
-                        )
-
-                        has_mutation_str = (
-                            check_for_mutation_ignore_cuda_graph_managed_tensor(
-                                gm,
-                                compiled_graph,
-                                static_input_idxs,
-                            )
-                        )
-                        has_mutation = has_mutation_str is not None
-
-                        if has_mutation:
-                            compiled_graph.disabled_cudagraphs_reason = has_mutation_str
-                    else:
-                        # Check mutation later to support cudagraph-managed tensors
-                        has_mutation = None
-
-                    cudagraph_tests = [
-                        (not has_mutation, "mutated inputs"),
-                        (not complex_memory_overlap_inputs, "complex memory overlap"),
-                        (
-                            all(
-                                isinstance(t, (torch.Tensor, torch.SymInt))
-                                for t in example_inputs
-                            ),
-                            "non-Tensor inputs",
-                        ),
-                    ]
-                    output = output_node(gm)
-                    # output args are tuple of first argument
-                    assert len(output.args) == 1
-                    stack_traces = [
-                        (
-                            arg.stack_trace
-                            if isinstance(arg, torch.fx.node.Node)
-                            else None
-                        )
-                        for arg in output.args[0]
-                    ]
-                    cudagraph_fail_reasons = [s for b, s in cudagraph_tests if not b]
-                    placeholders = tuple(get_placeholder_info(gm.graph))
-                    cudagraph_info = CudagraphCachedInfo(
-                        placeholders, stack_traces, cudagraph_fail_reasons
-                    )
-
-            compiled_graph.cudagraph_info = cudagraph_info
-            compiled_graph.inputs_to_check = inputs_to_check
-            compiled_graph.fx_kwargs = fx_kwargs
-            # TODO: should this be part of fx_kwargs
-            compiled_graph.boxed_forward_device_index = boxed_forward_device_index
+            compiled_graph.post_compile1(
```

Comment:
The difference between post_compile1 and post_compile2 is that post_compile1 happens pre serialization, and post_compile2 happens post serialization. They're both "post_compile", but I would honestly consider post_compile1 to really be "part of compile", if we believe that CompiledFxGraph is the final "compiled" product we get.

Reply:
post_compile1 doesn't actually happen pre serialization.  In https://github.com/pytorch/pytorch/pull/141689 we discover they are actually happening one after another and can be merged.